enhance and refactor: multiple scripts for better functionality

omarcopires opened the pull request:
# Description
1. Fixes a minor issue where, when emptying the balloons, they were transformed into 255 instead of being reset to 1.
2. Adjusts the usage of the "bathtub" in datapacks, in addition to the RLMap, ensuring better compatibility.
3. Makes minor aesthetic changes to enhance the code's readability.

Close #2923

## Type of change
  - [x] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)

## Checklist
  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

Commit:
fix: prevent multiple transformations of anniversary balloons

Commit:
fix: bath tube for custom data

Commit:
chore: vessels for custom data

Commit:
chore: dolls for custom data

Commit:
fix: replace SoulWarQuest.bagYouDesireItemId with BAG_YOU_DESIRE

Commit:
fix: ensure only one prize is awarded from surprise box

Commit:
refactor: simplify surprise bags logic and reduce code size

Commit:
refactor: improve readability and organization of bed modification kits

Commit:
fix: update guild war status logic

Commit:
refactor: simplify reward distribution in christmas bundle action

Commit:
chore: new items in core

Commit:
refactor: optimize spectator message handling and item transformation logic

Commit:
Merge branch 'main' into fix-balloons

dudantas commented on data-otservbr-global/lib/quests/soul_war.lua:
```diff
@@ -1,7 +1,6 @@
 SoulWarQuest = {
 	-- Item ids
 	-- Goshnar's Hatred
-	bagYouDesireItemId = 34109,
```
As I said in a previous pull request, this is unnecessary and breaks the encapsulation of the system. (besides being an unnecessary change)

omarcopires replied:
Nothing is breaking; on the contrary, we're simply reusing a constant that's already registered for other scripts.

![image](https://github.com/user-attachments/assets/6d8611b9-8a26-499f-a4a2-9283ce3a2cc8)

dudantas replied:
As I mentioned in a previous pull request, this proposed change is problematic, and I won’t accept it. Moving a constant, which is an intrinsic part of the quest's logic, outside of the quest file significantly reduces code cohesion. This not only complicates installation for other users, who would have to look for a constant in a non-intuitive location, but also greatly increases the risk of conflicts during operations like cherry-picks or merges.

Keeping the constant within the quest file is precisely to encapsulate its logic, ensuring that all necessary dependencies are present and clearly visible within the context of the quest. Moving this constant to another file breaks this encapsulation and can lead to errors or comprehension difficulties for those attempting to implement or update the quest on their servers.

Without this reversion, I won’t be able to merge your pull request.

Commit:
resolve conversation

Commit:
refactor: enhance VIP status handling and improve clarity

Commit:
Merge branch 'main' into fix-balloons

Commit:
fix: improve gold ranking display with formatted values and variable renaming

Changes to the gold ranking script to improve readability and formatting. Variable names were renamed to be more descriptive, and the FormatNumber function was used to display balance values in a formatted way. The database query result check was adjusted for better clarity.

Commit:
improvement: addMounts command to assign mounts to players

Commit:
improvement: /addaddon to support adding specific or all outfit addons

Commit:
Merge remote-tracking branch 'upstream/main' into fix-balloons

github-actions[bot] committed:
chore: move crushers to core

Commit:
Merge branch 'main' into fix-balloons

Commit:
fix: improve skill addition messages for clarity and correct pluralization

Commit:
Merge remote-tracking branch 'upstream/main' into fix-balloons

Commit:
remove duplicated script

Commit:
feat: gooey mass

Commit:
Merge branch 'main' into fix-balloons

Commit:
chore: move scripts to core

Commit:
refactor: combine findItemInInbox functions into one

Commit:
Merge remote-tracking branch 'upstream/main' into fix-balloons

Commit:
feat: mystery box

Commit:
revert changes

dudantas commented on data/scripts/talkactions/god/add_addon.lua:
```diff
@@ -1,45 +1,305 @@
-function addons.onSay(player, words, param)
-	-- create log
-	logCommand(player, words, param)
+local looktypes = {
```
This seems a bit redundant, as we already have the outfit list in the cpp. Wouldn't it be better to just "get" it here, rather than hardcoding these values again? It would avoid having another place to update hardcoded IDs.

https://github.com/opentibiabr/canary/blob/1b4de5d59ff5371c6d933747cb2f119f14a72978/src/creatures/appearance/outfit/outfit.hpp#L51

I saw that it was like this before, but since you are already modifying it here, I believe that the work is small to create a lua function exporting the cpp table.

```lua
int GameFunctions::luaGameGetOutfits(lua_State* L) {
	// Game.getOutfits()
	const auto &outfits = g_game().getOutfits();
	lua_createtable(L, outfits.size(), 0);
	int index = 0;
	for (const auto &outfit : outfits) {
		lua_createtable(L, 0, 2);
		setField(L, "id", outfit.first);
		setField(L, "name", outfit.second);
		lua_rawseti(L, -2, ++index);
	}
	return 1;
}
```

```lua
local looktypes = {}

for _, outfit in ipairs(outfits) do
    table.insert(looktypes, outfit.id) 
end
```

dudantas commented on data/scripts/talkactions/god/add_mount.lua:
```diff
@@ -16,23 +10,41 @@ function addOutfit.onSay(player, words, param)
+
+	local mountParam = string.trim(split[2])
+	if mountParam == "all" then
+		for mountId = 1, 231 do
```
Same here, but it's even easier, you can just export "getMounts().size()" and call it here

```cpp

int GameFunctions::luaGameGetMountsSize(lua_State* L) {
	// Game.getMountsSize()
	lua_pushnumber(L, g_game().getMounts().size());
	return 1;
}
```

dudantas reviewed (commented):
Here are a few suggestions, if you'd like to implement them. It was done this way before, but since you're already working on this part, I believe it wouldn't be too much additional effort.

Commit:
Merge remote-tracking branch 'upstream/main' into fix-balloons

Commit:
typo

Commit:
Merge branch 'main' into fix-balloons

Commit:
refactor: rename variables to camelCase and improve readability

Commit:
Merge branch 'main' into fix-balloons
